frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -28,22 +28,47 @@ const selectorReducer = (state: State, action: Action): State => {
     // User actions.
 
     case "ADD_PROJECTS": {
+      let newState = { ...state };
+
       // a given custom project may already exist in the group so don't trigger a state update for those duplicates
       const uniqueCustomProjects = action.payload.projects.filter(
-        (project: string) => !(project in state[action.payload.group])
+        (project: string) => !(project in newState[action.payload.group])
       );
       if (uniqueCustomProjects.length === 0) {
-        return state;
+        return newState;
       }
-      return {
```

Comment:
i would keep this as-is (assign to  newState instead of return), you don't want to return copied state unless the state was modified.

Reply:
also let's add a comment for this logic now that there's the additional step of adding upstreams and downstreams